mongodb/docs-worker-pool
# DOP-3484: Suppress incompatible types warnings for Atlas Admin API

### Ticket

[DOP-3484](https://jira.mongodb.org/browse/DOP-3484)

### Testing

To test locally:
1) Pull the changes from this branch.
2) Run `npm run build` for the OAS module.
3) Run the latest version of the snooty parser to parse `cloud-docs`.
4) Ensure your local copy of the `redoc` repo has the latest changes from the `main` branch, or from https://github.com/mongodb-forks/redoc/pull/6.
5) Run the command below from the OAS module directory for the `cloud-docs` repo. See [README](https://github.com/mongodb/docs-worker-pool/blob/19f9c26f450998e97f781701a6052b09885d5ffe/modules/oas-page-builder/README.md#installation) for more details, if needed. 
```
node dist/index.js --bundle </path/to/parser/output/bundle.zip> --output </path/to/directory/output> --redoc </path/to/local/redoc/cli/index.js> --repo </path/to/local/repo/directory> --site-url "https://www.mongodb.com/docs/atlas/"
```

The same can be done with `docs-app-services`.

### Notes

* Adds the new `ignoreIncompatibleTypes` option as a build option for Atlas-based builds. This will prevent the Atlas Admin API from emitting the "incompatible types" warnings in `cloud-docs` build logs.
* Updates the custom options passed to the CLI build command to fit the new syntax.
* Bumps the CLI version to tag v0.2.0, which contains the new option changes.
* Build logs show no "incompatible types" warnings:
   * [cloud-docs](https://workerpoolstaging-qgeyp.mongodbstitch.com/pages/job.html?collName=dotcom_queue&jobId=63d94c83f0636c6127b7afab)
   * [docs-app-services](https://workerpoolstaging-qgeyp.mongodbstitch.com/pages/job.html?collName=dotcom_queue&jobId=63d94c83f0636c6127b7afac) (as a sanity check)
